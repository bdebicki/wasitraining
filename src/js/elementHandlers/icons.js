export const iconClassBuilder = 'icon';
export const ICONS = {
	PLUS: `${iconClassBuilder}--plus`,
	CLOSE: `${iconClassBuilder}--close`,
	CLOSE_RECT: `${iconClassBuilder}--close__line`,
};
