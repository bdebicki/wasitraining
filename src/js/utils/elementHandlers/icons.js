const iconClassBuilder = 'icon';
const ICONS = {
  PLUS: `${iconClassBuilder}--plus`,
  CLOSE: `${iconClassBuilder}--close`,
  CLOSE_RECT: `${iconClassBuilder}--close__line`,
};

export {
  iconClassBuilder,
  ICONS,
};
