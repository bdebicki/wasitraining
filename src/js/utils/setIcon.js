import addSvgElement from './addSvgElement';

export default function setIcon(shape, classNames) {
	const settings = {
		children: shape.icon,
		classNames: [shape.className].concat(classNames),
		properties: [
			{ width: `${shape.width}px` },
			{ height: `${shape.height}px` },
			{ viewBox: `0 0 ${shape.width} ${shape.height}` },
		],
	};

	return addSvgElement('svg', settings);
}
