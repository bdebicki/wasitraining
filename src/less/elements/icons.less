@plusIconSize: 6px;
@closeIconSize: 10px;

.icon {
	&--plus {
		width: @plusIconSize;
		height: @plusIconSize;
	}
	&--close {
		width: @closeIconSize;
		height: @closeIconSize;

		&__line {
			transform: rotate(45deg);
			transform-origin: 50% 50%;
		}
	}
}