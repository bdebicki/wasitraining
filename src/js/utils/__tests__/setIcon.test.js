import setIcon from '../setIcon';
import icons from '../../helpers/iconsLibrary';

describe('tests for set icon', () => {
	it('it renders icon with single class name', () => {
		expect(setIcon(icons.close(), 'iconClose')).toMatchSnapshot();
	});
	it('it renders icon with multiple class names', () => {
		expect(setIcon(icons.close(), ['iconClose', 'additionalClass'])).toMatchSnapshot();
	});
});
