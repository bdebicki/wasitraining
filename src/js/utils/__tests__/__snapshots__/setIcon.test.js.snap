// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for set icon it renders icon with multiple class names 1`] = `
<svg
	class="icon--close iconClose additionalClass"
	height="10px"
	viewBox="0 0 10 10"
	width="10px"
>
	<rect
		class="icon--close__line"
		height="2"
		width="12"
		x="-1"
		y="4"
	/>
	<rect
		class="icon--close__line"
		height="12"
		width="2"
		x="4"
		y="-1"
	/>
</svg>
`;

exports[`tests for set icon it renders icon with single class name 1`] = `
<svg
	class="icon--close iconClose"
	height="10px"
	viewBox="0 0 10 10"
	width="10px"
>
	<rect
		class="icon--close__line"
		height="2"
		width="12"
		x="-1"
		y="4"
	/>
	<rect
		class="icon--close__line"
		height="12"
		width="2"
		x="4"
		y="-1"
	/>
</svg>
`;
